### ISSUE TITLE
[BUG] Issue with parsing -a/--addr

### ISSUE THREAD

**Issue by mikepietruszka** [ycd/dstp#12]
Hello,

I've noticed that I am unable to run the tests with `./dstp -a 8.8.8.8` on macOS (haven't tested on Linux yet). I've isolated the issue with how the flags are parsed in https://github.com/ycd/dstp/blob/1c9140cdbe66bf5245eca70cc98ab427d2e43865/config/config.go#L44

I was able to isolate it by adding `fmt.Println(values)` in `cmd/dstp/main.go` and seeing that `values` was empty:

```bash
$ make; sudo ./dstp -a 8.8.8.8
go build -o dstp cmd/dstp/main.go
Password:
opts.Addr value: 8.8.8.8
values variable value: []
address cannot be empty

Usage: dstp [OPTIONS] [ARGS]
Options:
        -a, --addr   <string>  The URL or the IP address to run tests against      [REQUIRED]
        -o, --out    <string>  The type of the output, either json or plaintext    [Default: plaintext]
        -c           <bool>    Run all the tests concurrently.                     [Default: false]
        -h, --help             Show this message and exit.
```

In order to run it, I have to update https://github.com/ycd/dstp/blob/1c9140cdbe66bf5245eca70cc98ab427d2e43865/config/config.go#L69 to remove the `values` size condition:

```bash
$ tail config/config.go
	if opts.Addr != "" { // && len(values) >= 1 {
                // opts.Addr = values[0]
        } else {
                fmt.Println(opts.Addr)
                return nil, fmt.Errorf("address cannot be empty")
        }

        return opts, nil
}
```
```bash
$ make; sudo ./dstp -a 8.8.8.8
go build -o dstp cmd/dstp/main.go
Password:
opts.Addr value: 8.8.8.8
values variable value: []
Ping: 329.766ms
DNS: resolving 8.8.8.8
TLS: certificate is valid for 60 more days
HTTPS: got 200 OK
```

Another issue I've noticed is that on macOS, you need to set `pinger.SetPrivileged(true)` on https://github.com/ycd/dstp/blob/1c9140cdbe66bf5245eca70cc98ab427d2e43865/pkg/ping/ping.go#L13 and https://github.com/ycd/dstp/blob/1c9140cdbe66bf5245eca70cc98ab427d2e43865/pkg/ping/ping.go#L41 and run `dstp` with `sudo`. I suppose that it's the same issue as on Linux.

go version:
```
$ go version
go version go1.17.2 darwin/amd64
```
OS:
```
$ uname -a
Darwin 20.6.0 Darwin Kernel Version 20.6.0: Tue Oct 12 18:33:42 PDT 2021; root:xnu-7195.141.8~1/RELEASE_X86_64 x86_64
```

**Comment by ycd** [comment]
Hey @mikepietruszka, thanks reporting in a very detailed fashion, much appreciated.  

I thought using `flagset.Args()` for addr could be better for simplicty, looks like i forgot to add logic for that if addr is given by `--addr`flag. 

> Another issue I've noticed is that on macOS, you need to set pinger.SetPrivileged(true) on

I'm not sure about this, are you getting an error if you run it without `pinger.SetPrivileged(true)` ?

**Comment by mikepietruszka** [comment]
@ycd, groovy, latest changes in [7768b61459f5991686ba05539599007cbd618d8d](https://github.com/ycd/dstp/commit/7768b61459f5991686ba05539599007cbd618d8d#diff-fe44f09c4d5977b5f5eaea29170b6a0748819c9d02271746a20d81a5f3efca17) make it work with `-a` and without that flag:

```bash
[20:48:49 ~/code/go/src/dstp g:main]
$ make; sudo ./dstp 8.8.8.8
go build -o dstp cmd/dstp/main.go
Password:
Ping: 37.945666ms
DNS: resolving 8.8.8.8
SystemDNS: resolving 8.8.8.8
TLS: certificate is valid for 59 more days
HTTPS: got 200 OK
```
```bash
$ make; sudo ./dstp -a 8.8.8.8
go build -o dstp cmd/dstp/main.go
Password:
Ping: 12.568667ms
DNS: resolving 8.8.8.8
SystemDNS: resolving 8.8.8.8
TLS: certificate is valid for 59 more days
HTTPS: got 200 OK
```

As for as inability to run without `sudo`, I don't get any errors - the code just hangs. I can get an abbreviated `dtruss` output. I don't believe it gets to `pinger.Count = 3` from my debugging.

Either way, huge thanks for this tool. I saw it on HN last night and I must admit that I've been looking for some comprehensive networking test tool. So far this hits the bill. Great job and hope I can contribute in some ways.

**Comment by ycd** [comment]
Hmm even though i was testing the executable manually on MacOS, Fedora and Ubuntu, i wasn't able to find the reason behind tests were hanging on CI, that explained the reason behind that. 

Then i found this issue in the library that is used in dstp: https://github.com/go-ping/ping/issues/180.

This seems like a library problem, i should temporarily add something like "requires usage of sudo in XX" somewhere in README.

**Comment by ycd** [comment]
Hey again @mikepietruszka, just released v0.3.1 which includes fallback option for dstp(tries to execute ping command under the hood) to work in non-privileged cases and added timeout flag for specifying ping timeouts.
